network-mock/src/network.rs

```diff
@@ -455,7 +455,126 @@ impl Network for MockNetwork {
         );
 
         if self.is_connected.load(Ordering::SeqCst) {
-            if let Some(topic) = hub.get_topic(topic_name) {
+            if let Some(topic) = hub.get_topic(&topic_name) {
```

Comment:
I'd have expected to re-use some of the already existing functionality to not duplicate code